File: test/language/templates/python/InstantiateViaSingleImportTest.py
import zserio

import Templates

class InstantiateViaSingleImportTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateViaSingleImport = self.api.InstantiateViaSingleImport(
            self.api.pkg.U32(13),
            self.api.pkg.Test_string("test")
        )

        writer = zserio.BitStreamWriter()
        instantiateViaSingleImport.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateViaSingleImport = self.api.InstantiateViaSingleImport()
        readInstantiateViaSingleImport.read(reader)
        self.assertEqual(instantiateViaSingleImport, readInstantiateViaSingleImport)

File: test/language/templates/cpp/InstantiateViaSingleImportTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_via_single_import/InstantiateViaSingleImport.h"

namespace templates
{
namespace instantiate_via_single_import
{

TEST(InstantiateViaSingleImportTest, readWrite)
{
    InstantiateViaSingleImport instantiateViaSingleImport;
    instantiateViaSingleImport.setTest32(pkg::U32{13});
    instantiateViaSingleImport.setTestStr(pkg::Test_string{"test"});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateViaSingleImport.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateViaSingleImport readInstantiateViaSingleImport(reader);

    ASSERT_TRUE(instantiateViaSingleImport == readInstantiateViaSingleImport);
}

} // namespace instantiate_via_single_import
} // namespace templates
